core/unit_test/tools/TestWithoutInitializing.cpp

```diff
@@ -47,25 +47,43 @@
 
 #include "include/ToolTestingUtilities.hpp"
 
+template <typename InputView, typename OutputView>
+void check_host_mirror(const InputView&, const OutputView&) {
+  using InputSpace  = typename InputView::memory_space;
+  using OutputSpace = typename OutputView::memory_space;
+  if constexpr (Kokkos::SpaceAccessibility<Kokkos::HostSpace,
+                                           InputSpace>::accessible)
+    static_assert(std::is_same_v<OutputSpace, InputSpace>);
+  else
+    static_assert(std::is_same_v<OutputSpace, Kokkos::HostSpace>);
+}
+
 TEST(kokkosp, create_mirror_no_init) {
   using namespace Kokkos::Test::Tools;
   listen_tool_events(Config::DisableAll(), Config::EnableKernels());
   Kokkos::View<int*, Kokkos::DefaultExecutionSpace> device_view("device view",
                                                                 10);
   Kokkos::View<int*, Kokkos::HostSpace> host_view("host view", 10);
+  using MemorySpace = typename Kokkos::DefaultExecutionSpace::memory_space;
 
   auto success = validate_absence(
       [&]() {
         auto mirror_device =
             Kokkos::create_mirror(Kokkos::WithoutInitializing, device_view);
+        check_host_mirror(device_view, mirror_device);
         auto mirror_host =
             Kokkos::create_mirror(Kokkos::WithoutInitializing,
                                   Kokkos::DefaultExecutionSpace{}, host_view);
+        static_assert(
+            std::is_same_v<decltype(mirror_host)::memory_space, MemorySpace>);
         auto mirror_device_view = Kokkos::create_mirror_view(
             Kokkos::WithoutInitializing, device_view);
+        check_host_mirror(device_view, mirror_device_view);
         auto mirror_host_view = Kokkos::create_mirror_view(
             Kokkos::WithoutInitializing, Kokkos::DefaultExecutionSpace{},
             host_view);
+        static_assert(std::is_same_v<decltype(mirror_host_view)::memory_space,
+                                     MemorySpace>);
```

Comment:
This unit test is about using the tools hooks to check that we allocate, fence, or launch kernels only when appropriate.  The compile time testing of `create_mirror` does not really belong there.  Did you consider moving these elsewhere and asserting in a more systematic way?

Reply:
For the container tests, all these overloads are tested in the `TestWithoutInitializing` file. Thus, it feels like a good choice to extend those tests slightly instead of duplicating the setup somewhere else. By extension, it appears sensible to me to add the compile-time tests to a corresponding file in `core`.